src/compiler/checker.ts

```diff
@@ -15930,6 +15931,9 @@ namespace ts {
                             getTypeFromTypeNodeWorker(<TypeNode>node) === tp; // use worker because we're looking for === equality
                     case SyntaxKind.TypeQuery:
                         return true;
+                    case SyntaxKind.MethodDeclaration:
+                    case SyntaxKind.MethodSignature:
+                        return (!(node as FunctionLikeDeclaration).type && !!(node as FunctionLikeDeclaration).body) || !!forEachChild(node, containsReference);

```

Comment:
@weswigham I'm fixing #44292 which was indirectly caused by the changes in this PR. Now I'm looking at this code and I'm not really sure why you return true if there's no type annotation but a function body, but otherwise you visit every node in the function body.

Reply:
Before your changes in this PR, we'd only call `isTypeParameterPossiblyReferenced` on something that is a type. Now we can end up crawling through method bodies which this code wasn't intended to do.